### PR TITLE
Merge threads.h into threading/thread.h

### PR DESCRIPTION
**Pull request by ShadowNinja**
Now we only have one file for thread management.

### CODE REVIEW

**Comment by kwolekr** [comment]
I'm not on board with changing all of the threadid_ts to something else.

File: src/threading/thread.h [hunk 1]
```diff
@@ -158,9 +203,9 @@ class Thread {
 	Mutex m_mutex;
 
 #if !USE_CPP11_THREADS
```
Comment (Zeno-) [inline src/threading/thread.h]: This should be #if**n**def USE_CPP11_THREADS, or (better) combined with the #if USE_CPP11_THREADS on line 220

**Comment by nerzhul** [comment]
:-1: for renaming threadid_t else ok

File: src/threading/thread.h (hunk 1 above)
Comment (ShadowNinja) [inline src/threading/thread.h]: I didn't change this line in this PR.

**Comment by kwolekr** [comment]
Just a friendly ping on this PR.

I am on board with consolidating the threads code into a single file, but not the threadid_t rename.  It seems unnecessary and setting that kind of precedent means we'd have to change content_t as well which is another big find & replace operation that doesn't actually provide tangible benefit to the project.

Could you split the two changes into distinct PRs?  Or would you prefer that I close this one out for you and open a new one that just merges the threading files instead?

**Comment by nerzhul** [comment]
I have the same opinion as @kwolekr on this subject

**Comment by ShadowNinja** [comment]
@kwolekr: Changed back to `global_lowercase_underscore_t` style.

**Comment by nerzhul** [comment]
i'm closing this, we are near feature freeze and this part will be refactored after release to use pure C++11

**Pull request closed without merging**